src/Controller/OpenyMemberships.php

```diff
@@ -65,13 +82,52 @@ public function __construct(EntityTypeManagerInterface $entity_type_manager, Con
    */
   public static function create(ContainerInterface $container) {
     return new static(
+      $container->get('entity.query'),
       $container->get('entity_type.manager'),
       $container->get('config.factory'),
       $container->get('commerce_cart.cart_provider'),
       $container->get('current_user')
     );
   }
 
+  /**
+   * Get Ages Groups and related products.
+   */
+  public function getAgesGroupsInfo(Request $request) {
+    $data = [];
+    $tids = $this->entityQuery
+      ->get('taxonomy_term')
+      ->condition('vid', 'memberships_ages_groups')
+      ->sort('weight', 'ASC')
+      ->execute();
+    $terms = $this->entityTypeManager->getStorage('taxonomy_term')->loadMultiple($tids);
+    foreach ($terms as $tid => $term) {
+      if (
+        $product_ids = $this->entityQuery
+        ->get('commerce_product')
+        ->condition('field_om_total_free.target_id', $tids, 'IN')
+        ->condition('status', 1)
+        ->execute()
+      ) {
+        $products = $this->entityTypeManager->getStorage('commerce_product')->loadMultiple($product_ids);
```

Comment:
There is no need to load storage CommeProduct::loadMultiple works as well and more readable.

Reply:
Product::loadMultiple